app/src/main/java/com/mapbox/maps/testapp/examples/CameraPredefinedAnimatorsActivity.kt

```diff
@@ -132,48 +139,83 @@ class CameraPredefinedAnimatorsActivity : AppCompatActivity() {
     cameraAnimationsPlugin.unregisterAnimators(*currentAnimators)
   }
 
+  private fun prepareAnimationOptions(duration: Long): MapAnimationOptions {
+    return mapAnimationOptions {
+      duration(duration)
+      animatorListener(object : AnimatorListenerAdapter() {
+
+        override fun onAnimationStart(animation: Animator?) {
+          super.onAnimationStart(animation)
+          runOnUiThread {
+            buttonCancel.visibility = View.VISIBLE
+          }
+        }
+
+        override fun onAnimationEnd(animation: Animator?) {
+          super.onAnimationEnd(animation)
+          runOnUiThread {
+            buttonCancel.visibility = View.INVISIBLE
+          }
+        }
+
+        override fun onAnimationCancel(animation: Animator?) {
+          super.onAnimationCancel(animation)
+          runOnUiThread {
+            buttonCancel.visibility = View.INVISIBLE
+          }
+        }
+      })
+    }
+  }
+
   private fun playAnimation(itemId: Int) {
     stopAnimation()
     when (itemId) {
       R.id.menu_action_jump_to -> mapboxMap.setCamera(START_CAMERA_POSITION)
-      R.id.menu_action_ease_to -> mapboxMap.easeTo(
-        EASE_TO_TARGET_CAMERA_POSITION,
-        mapAnimationOptions {
-          duration(2000)
+      R.id.menu_action_ease_to ->
+        mapboxMap.easeTo(
+          EASE_TO_TARGET_CAMERA_POSITION,
+          prepareAnimationOptions(2000)
+        )?.let {
+          runningCancelableAnimator = it
         }
-      )
-      R.id.menu_action_fly_to -> mapboxMap.flyTo(
-        EASE_TO_TARGET_CAMERA_POSITION,
-        mapAnimationOptions {
-          duration(2000)
+      R.id.menu_action_fly_to ->
+        mapboxMap.flyTo(
+          EASE_TO_TARGET_CAMERA_POSITION,
+          prepareAnimationOptions(2000)
+        )?.let {
```

Comment:
why the mapboxMap.flTo() returns a nullable?

Reply:
Good catch, should not be nullable.